### PR TITLE
Helm: allow the topologySpreadConstraint strategy to list namespaces

### PR DESCRIPTION
**Pull request by AmoVanB**
The `topologySpreadConstraint` strategy tries to list namespaces but the `ClusterRole` of the Helm chart does not allow it to. 

This PR adds this rule to the `ClusterRole` when the `topologySpreadConstraint` strategy is enabled.

### CODE REVIEW

**Review by damemi — APPROVED** [review]
/ok-to-test
/approve
thanks for catching this!

**Comment by damemi** [comment]
@AmoVanB looks like there is a syntax error in your change, not sure what the fix is for helm here:
```
 helm lint ./charts/descheduler
==> Linting ./charts/descheduler
[ERROR] templates/: template: descheduler/templates/clusterrole.yaml:15:14: executing "descheduler/templates/clusterrole.yaml" at <.Values.deschedulerPolicy.strategies.RemovePodsViolatingTopologySpreadConstraint.enabled>: nil pointer evaluating interface {}.enabled 
```

File: charts/descheduler/templates/clusterrole.yaml [hunk 1]
```diff
@@ -12,6 +12,11 @@ rules:
 - apiGroups: [""]
   resources: ["nodes"]
   verbs: ["get", "watch", "list"]
+{{- if .Values.deschedulerPolicy.strategies.RemovePodsViolatingTopologySpreadConstraint.enabled }}
```
Comment (seanmalloy) [inline charts/descheduler/templates/clusterrole.yaml:15]: In my opinion the conditional is not needed. The namespace list permission can just always be included. It is already included in file `kubernetes/base/rbac.yaml`.

**Review by seanmalloy — CHANGES_REQUESTED** [review]

File: charts/descheduler/templates/clusterrole.yaml (hunk 1 above)
Comment (seanmalloy) [reply charts/descheduler/templates/clusterrole.yaml:15]: https://github.com/kubernetes-sigs/descheduler/blob/70d1fadae750207acd6f7f3db07f868c2624b368/kubernetes/base/rbac.yaml#L13-L15

File: charts/descheduler/templates/clusterrole.yaml [hunk 2]
```diff
@@ -12,6 +12,11 @@ rules:
...
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["list"]
```
Comment (seanmalloy) [inline charts/descheduler/templates/clusterrole.yaml:18]:
```suggestion
  verbs: ["get", "list"]
```

File: charts/descheduler/templates/clusterrole.yaml (hunk 1 above)
Comment (AmoVanB) [reply charts/descheduler/templates/clusterrole.yaml:15]: sounds good to me, updating

**Review by seanmalloy — APPROVED** [review]

**Pull request merged**